app/containers/MessageBox/EmojiSearchbar.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useState } from 'react';
+import { View, Text, Pressable, TextInput, FlatList } from 'react-native';
+
+import { FormTextInput } from '../TextInput/FormTextInput';
+import { useTheme } from '../../theme';
+import I18n from '../../i18n';
+import { CustomIcon } from '../CustomIcon';
+import { IEmoji } from '../../definitions';
+import shortnameToUnicode from '../../lib/methods/helpers/shortnameToUnicode';
+import CustomEmoji from '../EmojiPicker/CustomEmoji';
+import styles from './styles';
+import { useFrequentlyUsedEmoji } from '../EmojiPicker';
+
+const BUTTON_HIT_SLOP = { top: 15, right: 15, bottom: 15, left: 15 };
+const EMOJI_SIZE = 30;
+const DEFAULT_EMOJIS = ['clap', '+1', 'heart_eyes', 'grinning', 'thinking_face', 'smiley'];
+interface IEmojiSearchbarProps {
+	openEmoji: () => void;
+	onChangeText: (value: string) => void;
+	emojis: IEmoji[];
+	onEmojiSelected: (emoji: IEmoji) => void;
+	baseUrl: string;
+}
+
+interface ICustomEmoji {
+	name: string;
+	extension: string;
+}
+
+const Emoji = React.memo(({ emoji, baseUrl }: { emoji: IEmoji; baseUrl: string }) => {
+	const { colors } = useTheme();
+	if (emoji.isCustom || emoji.name) {
+		return <CustomEmoji style={{ height: EMOJI_SIZE, width: EMOJI_SIZE, margin: 4 }} emoji={emoji} baseUrl={baseUrl} />;
+	}
+	return (
+		<Text style={[styles.searchedEmoji, { fontSize: EMOJI_SIZE, color: colors.backdropColor }]}>
+			{shortnameToUnicode(`:${emoji}:`)}
+		</Text>
+	);
+});
+
+const EmojiSearchbar = React.forwardRef<TextInput, IEmojiSearchbarProps>(
+	({ openEmoji, onChangeText, emojis, onEmojiSelected, baseUrl }, ref) => {
+		const { colors } = useTheme();
+		const [searchText, setSearchText] = useState<string>('');
+		const [frequentlyUsedEmojis, setFrequentlyUsed] = useState<(string | ICustomEmoji)[]>();
+		const { frequentlyUsed, loaded } = useFrequentlyUsedEmoji();
+
+		useEffect(() => {
```

Comment:
Wouldn't it be better to put this logic inside the emoji hook?

Reply:
I thought of doing so but this logic doesn't have any relation with frequently used emojis, also we don't want this behavior at other place so maybe it would not make sense to merge the login IMO.

Reply:
You're on the correct path, but still not quite right.
Your custom hooks is already returning the data you need.
You don't need an extra `useEffect` with `loaded` as dependency.

Just use `loaded` and `frequentlyUsed` to render what you need.

Reply:
Are you going to fix this?

Reply:
Fixed!